File: MethodsAsync/trycatch.js
function soma(a, b) {
    try {
        if (typeof a !== 'number' || typeof b !== 'number') {
            throw new Error('Os argumentos devem ser números');
        }
        
        return a + b;
    } catch (error) {
        console.error('Ocorreu um erro:', error.message);
    }
}

// Exemplo de uso:
console.log(soma(2, 3)); // Output: 5
console.log(soma(2, '3')); // Output: Ocorreu um erro: Os argumentos devem ser números

File: MethodsAsync/trycatch.lua
-- Função de soma com try-catch
function soma(a, b)
    local status, resultado = pcall(function()
        return a + b
    end)

    if status then
        return resultado
    else
        print("Erro ao executar a soma:", resultado)
        return nil
    end
end

-- Exemplo de uso da função
local resultado = soma(10, 5)
if resultado then
    print("Resultado da soma:", resultado)
end
